scipy/stats/_multivariate.py

```diff
@@ -5794,64 +5843,40 @@ class dirichlet_multinomial_gen(multi_rv_generic):
 
     Get the PMF
 
-    >>> import numpy as np
-    >>> alpha = np.array([3, 4, 5])
-    >>> x = np.array([1, 2, 3])
-    >>> dirichlet_multinomial.pmf(alpha, x)
+    >>> n = 6  # number of trials
+    >>> alpha = [3, 4, 5]  # concentration parameters
+    >>> x = [1, 2, 3]  # counts
+    >>> dirichlet_multinomial.pmf(x, alpha, n)
     0.08484162895927604
 
+    If the sum of category counts does not equal the number of trials,
+    the probability mass is zero.
+
+    >>> dirichlet_multinomial.pmf(x, alpha, n=7)
+    0.0
+
     Get the log of the PMF
 
-    >>> alpha = np.array([3, 4, 5])
-    >>> x = np.array([1, 2, 3])
-    >>> dirichlet_multinomial.logpmf(alpha, x)
+    >>> dirichlet_multinomial.logpmf(x, alpha, n)
     -2.4669689491013327
 
     Get the mean
 
-    >>> alpha = np.array([3, 4])
-    >>> n = 3
     >>> dirichlet_multinomial.mean(alpha, n)
-    array([1.28571429, 1.71428571])
+    array([1.5, 2. , 2.5])
 
     Get the variance
 
-    >>> alpha = np.array([1, 1, 1])
-    >>> n = 9
     >>> dirichlet_multinomial.var(alpha, n)
-    array([6.0, 6.0, 6.0])
+    array([1.55769231, 1.84615385, 2.01923077])
 
     Get the covariance
 
-    >>> alpha = np.array([1, 1, 1])
-    >>> n = 9
     >>> dirichlet_multinomial.cov(alpha, n)
     array([[6.0, -3.0, -3.0],
            [-3.0, 6.0, -3.0],
            [-3.0, -3.0, 6.0]])
 
-    The functions support broadcasting, under the convention that the
-    vector parameters are interpreted as if each row along the last
-    axis is a single object. For instance:
-
-    >>> alpha = np.array([[1, 2, 3], [4, 5, 6]])
-    >>> x = np.array([[1, 2, 3], [4, 5, 6]])
-    >>> dirichlet_multinomial.pmf(alpha, x)
-    array([0.06493506, 0.02626937])
-
-    This broadcasting also works for ``cov``, where the output objects are
-    square matrices of size ``m.shape[-1]``. For example:
-
-    >>> alpha = np.array([1, 1, 1])
-    >>> n = np.array([9, 9])
-    >>> dirichlet_multinomial.cov(alpha, n)
-    array([[[6.0, -3.0, -3.0],
-            [-3.0, 6.0, -3.0],
-            [-3.0, -3.0, 6.0]],
-           [[6.0, -3.0, -3.0],
-            [-3.0, 6.0, -3.0],
-            [-3.0, -3.0, 6.0]]])
-
```

Comment:
As much work as this PR is taking, we don't want to bother with broadcasting. That's a whole other can of worms. Let's start with the basics.